src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -60,9 +60,11 @@ class RelayDefaultNetworkLayer {
       } else {
         request.resolve({response: payload.data});
       }
-    }).catch(
-      error => request.reject(error)
-    );
+      return request;
+    }).catch(error => {
+      request.reject(error);
+      return request;
+    });
```

Comment:
What do you think about closing this PR and moving forward with the one to update the example?